pkg/vault/contracts/Router.sol

```diff
@@ -97,7 +97,7 @@ contract Router is IRouter, RouterCommon, ReentrancyGuardTransient {
                 // Transfer WETH from the Router to the Vault.
                 _weth.transfer(address(_vault), amountIn);
                 _vault.settle(_weth, amountIn);
-            } else {
+            } else if (amountIn > 0) {
```

Comment:
how about `if (amountIn == 0) break;` above instead? Would also apply to eth wraps.